File: src/com/leozhi/topic771/Solution03.java
package com.leozhi.topic771;

/**
 * @author leozhi
 * indexOf()
 * 通过
 * 1ms
 */
public class Solution03 {
    public int numJewelsInStones(String J, String S) {
        int res = 0;
        for (char c : S.toCharArray()) {
            if (J.indexOf(c) != -1) {
                res++;
            }
        }
        return res;
    }
}

File: src/com/leozhi/topic387/Solution03.kt
package com.leozhi.topic387

/**
 * @author leozhi
 * indexOf() == lastIndexOf()
 * 通过
 * 260ms
 */
class Solution03 {
    fun firstUniqChar(s: String): Int {
        var res = -1
        for (ch in 'a'..'z') {
            val index = s.indexOf(ch)
            if (index != -1 && s.lastIndexOf(ch) == index) {
                res = if ((res == -1 || index < res)) index else res
            }
        }
        return res
    }
}